mars/services/task/supervisor/preprocessor.py

```diff
@@ -203,6 +205,11 @@ def tile(self, tileable_graph: TileableGraph) -> Iterable[ChunkGraph]:
                     optimize_chunk_graph(chunk_graph)
                 )
             yield chunk_graph
+        # Extract ids of tileables whose op need func cleanup.
+        for t in tileable_graph:
+            if hasattr(t.op, "func_key") and t.op.func_key is not None:
```

Comment:
Do we need to check all the tileables again? You have checked the tileables a few lines above.